backend/src/Designer/Services/Implementation/ResourceRegistryService.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Altinn.ApiClients.Maskinporten.Interfaces;
+using Altinn.ApiClients.Maskinporten.Models;
+using Altinn.Studio.Designer.Configuration;
+using Altinn.Studio.Designer.Models;
+using Altinn.Studio.Designer.Services.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+
+namespace Altinn.Studio.Designer.Services.Implementation
+{
+    public class ResourceRegistryService : IResourceRegistry
+    {
+        private readonly HttpClient _httpClient;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IMaskinportenService _maskinPortenService;
+        private readonly IClientDefinition _maskinportenClientDefinition;
+        private readonly PlatformSettings _platformSettings;
+
+        public ResourceRegistryService()
+        {
+
+        }
+
+        public ResourceRegistryService(HttpClient httpClient, IHttpClientFactory httpClientFactory, IMaskinportenService maskinportenService, IClientDefinition maskinPortenClientDefinition, PlatformSettings platformSettings)
+        {
+            _httpClient = httpClient;
+            _httpClientFactory = httpClientFactory;
+            _maskinPortenService = maskinportenService;
+            _maskinportenClientDefinition = maskinPortenClientDefinition;
+            _platformSettings = platformSettings;
+        }
+
+        public async Task<ActionResult> PublishServiceResource(ServiceResource serviceResource, string env = null)
+        {
+            TokenResponse tokenResponse = await GetBearerTokenFromMaskinporten();
+            
+            string fullResourceRegistryUrl;
+            if (env != null)
+            {
+                fullResourceRegistryUrl = $"{string.Format(_platformSettings.ResourceRegistryEnvBaseUrl, env)}{_platformSettings.ResourceRegistryUrl}";
+                tokenResponse = await _maskinPortenService.ExchangeToAltinnToken(tokenResponse, env);
+            }
+            else
+            {
+                fullResourceRegistryUrl = $"{string.Format(_platformSettings.ResourceRegistryDevBaseUrl, env)}{_platformSettings.ResourceRegistryUrl}";
+            }
+
+            string serviceResourceString = JsonConvert.SerializeObject(serviceResource);
+            _httpClientFactory.CreateClient("myHttpClient");
+            _httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", tokenResponse.AccessToken);
+
+            HttpResponseMessage response = await _httpClient.PostAsync(fullResourceRegistryUrl, new StringContent(serviceResourceString, Encoding.UTF8, "application/json"));
```

Comment:
En fremtidig utvidelse her er å lage støtte for put for eksisterende ressurser. Hvor enkelt er det å gjøre et kall for å sjekke om den finnes allerde? Burde holde med en get mot ressurs registeret. Slik at man kan oppdatere

Reply:
Har endret på resourceregistry siden slik at det returneres "409 Conflict" derfra om ressursen allerede eksisterer og deretter returneres samme 409 fra publish endepunktet. Opprettet task på å opprette et PUT-endepunkt.